bot/models/note_book.py

```diff
@@ -0,0 +1,51 @@
+from .note import Note
+from datetime import datetime
+from collections import UserDict
+
+class NoteBook(UserDict):
+
+	def add_note(self, note: Note):
+		self.data[note.id] = note
+
+	def edit_note(self, index, new_text):
+		if 0 <= index < len(self.data):
+			self.data[index].text = new_text
+
+	def delete_note (self, index):
+		if 0 <= index < len(self.data):
+			del self.data[index]
+
+	def search_notes(self, keyword):
+		return [note for note in self.data if keyword in note.text]
+
+	def search_by_tag(self, tag):
+		return [note for note in self.data if tag in note.tags]
+
+	def sort_by_tag(self):
+		return sorted(self.data, key=lambda note: note.tags)
+	
+	def __str__(self) -> str:
```

Comment:
Implement the output in a table format using the 'rich' library.
Refer to the get_upcoming_birthdays method in address_book.py for guidance.

Reply:
done